sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -2348,11 +2348,56 @@ def _import_zip_as_action_package(
                 log.exception(msg)
                 return ActionResult.make_failure(msg).as_dict()
 
+    def m_drop_data_source(
+        self,
+        datasource: DatasourceInfoTypedDict,
+        data_server_info: DataServerConfigTypedDict,
+    ) -> ActionResultDict:
+        from sema4ai_code.data.data_server_connection import DataServerConnection
+
+        http = data_server_info["api"]["http"]
+        auth = data_server_info["auth"]
+        user = auth["username"]
+        password = auth["password"]
+
+        connection = DataServerConnection(
+            http_url=f"http://{http['host']}:{http['port']}",
+            http_user=user,
+            http_password=password,
+        )
+
+        name = datasource.get("name")
+        engine = datasource["engine"]
+        created_table = datasource.get("created_table")
+        model_name = datasource.get("model_name")
+
+        query = ""
+        if engine == "files":
```

Comment:
I think you should verify if what they want to drop exists and return a "success" result with a message saying that nothing happened because it wasn't there (it's strange to receive a message saying that it was successfully dropped when it didn't even exist).

-- p.s.: don't do this now, let's merge and do it in a separate PR.